### DISCUSSION TITLE
Initialize a value when a future provider was complete to load.

### DISCUSSION THREAD

**Discussion by billbull21** [rrousselGit/riverpod#1328] [Q&A]
excuse me,,
future provider is great, very helpful and just need only one call to API's.
but, currently i have a problem using future provider. the problem comes out when i try to initialize a (other) StateProvider inside my future provider.
first load, ok. but, when i come back to the screen.. which is the future provider start to fetch the cache data. my state provider couldn't initialize.

and one thing,, how can i update a stateprovider in a runtime ?
thanks.

**Comment by rrousselGit** [comment]
Could you clarify the problem you faced, ideally with a small example

**Reply by billbull21** [reply]
so,, i have a StateProvider that i want to change the data precisely after other FutureProvider was successfully have a data.
on first load is success, but when the FutureProvider got disposed and i try to open the screen again, the FutureProvider actually wouldn't to fetch the data from the remote server again, but it fetch from the cache.
and the problem in a sort way is, i couldn't "reinitialize" my state provider by rely on FutureProvider.

i got the temporary solution, but i don't know it is good or bad.
i put ref.read the future provider in initstate to "reinitialize" my StateProvider and also i put ref.watch in build method and update the data inside future provider for the first initialize.

btw, thanks for the response and help.
